File: adventofcode-2024/src/main/java/nl/mvdr/adventofcode/adventofcode2024/day16/Part2.java
package nl.mvdr.adventofcode.adventofcode2024.day16;

import java.util.stream.Stream;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import nl.mvdr.adventofcode.solver.LongSolver;

public class Part2 implements LongSolver {

    private static final Logger LOGGER = LoggerFactory.getLogger(Part2.class);

    @Override
    public long solve(Stream<String> lines) {
        return Maze.parse(lines.toList()).countBestPathTiles();
    }

    public static void main(String[] args) {
        var instance = new Part2();

        var result = instance.solve("input-day16-2024.txt");

        LOGGER.info(result);
    }
}
 

File: adventofcode-2025/src/main/kotlin/nl/mvdr/adventofcode/adventofcode2025/day07/Part2.kt
package nl.mvdr.adventofcode.adventofcode2025.day07

import io.github.oshai.kotlinlogging.KotlinLogging
import nl.mvdr.adventofcode.solver.FunctionSolver

private val logger = KotlinLogging.logger{}

fun solvePart2(lines: Sequence<String>) = parseManifold(lines.toList()).countTimelines()

fun main() {
    val result = FunctionSolver(::solvePart2).solve("input-day07.txt")
    logger.info { result }
}
